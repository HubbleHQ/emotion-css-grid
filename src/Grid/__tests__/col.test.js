import React from 'react';
import { shallow } from 'enzyme';

import { col } from '..';

describe('Grid : col(n) method', () => {
  it('should render via Emotion as CSS when passed as a css prop to a React component', () => {
    const test = col(2);
    const wrapper = shallow(<div css={test} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should return an Emotion object when passed a valid number', () => {
    const test = col(2);
    expect(typeof test).toBe('object');
  });
  it('should throw an error when passed a non-number', () => {
    const testFunction = () => {
      const test = col('oops');
    }
    expect(testFunction).toThrowErrorMatchingSnapshot();
  });
  it('should throw an error when passed a number that is less than 1', () => {
    const testFunction = () => {
      const test = col(0);
    }
    expect(testFunction).toThrowErrorMatchingSnapshot();
  });
  it('should throw an error when passed a number that is greater than the max number of grid columns', () => {
    const testFunction = () => {
      const test = col(75);
    }
    expect(testFunction).toThrowErrorMatchingSnapshot();
  });
});
