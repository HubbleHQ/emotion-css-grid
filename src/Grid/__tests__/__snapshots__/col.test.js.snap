// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid : col(n) method should render via Emotion as CSS when passed as a css prop to a React component 1`] = `
<div
  css={
    Object {
      "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInJlc3BvbnNpdmVXaWR0aC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFjeUMiLCJmaWxlIjoicmVzcG9uc2l2ZVdpZHRoLmpzIiwic291cmNlc0NvbnRlbnQiOlsiLyoqIEBqc3gganN4ICovXG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcblxuaW1wb3J0IGNvbmZpZyBmcm9tICcuL0dyaWRDb25maWcnO1xuXG5jb25zdCB7IG1vYmlsZUJyZWFrcG9pbnQgfSA9IGNvbmZpZztcblxuLyogQHBhcmFtIHtudW1iZXJ9IGNvbHVtbiB3aWR0aCBleHByZXNzZWQgYXMgYSBwZXJjZW50YWdlXG4gKiBUaGUgcmVzcG9uc2l2ZVdpZHRoIENTUyBkZXNjcmliZXMgaG93IGNvbHVtbnMgc2hvdWxkXG4gKiBhbHdheXMgY29sbGFwc2UgdG8gYSBzaW5nbGUtY29sdW1uIGxheW91dCAoMTAwJSB3aWR0aClcbiAqIG9uIG1vYmlsZSBzaXplcy4gQXQgYW55IHNpemUgYWJvdmUgYSBtb2JpbGUgcmFuZ2UsXG4gKiB0aGUgY29sdW1uJ3Mgd2lkdGggd2lsbCBiZSBzZXQgYnkgd2hhdGV2ZXIgbnVtZXJpY1xuICogcGVyY2VudGFnZSB2YWx1ZSBhcmd1bWVudCB0aGF0IGlzIHBhc3NlZCBpbi5cbiAqL1xuY29uc3QgcmVzcG9uc2l2ZVdpZHRoID0gd2lkdGhWYWx1ZSA9PiBjc3NgXG4gIHdpZHRoOiAxMDAlO1xuICBAbWVkaWEgKG1pbi13aWR0aDogJHttb2JpbGVCcmVha3BvaW50fXB4KSB7XG4gICAgd2lkdGg6ICR7d2lkdGhWYWx1ZX0lO1xuICB9XG5gO1xuXG5leHBvcnQgZGVmYXVsdCByZXNwb25zaXZlV2lkdGg7XG4iXX0= */",
      "name": "18dk701",
      "next": undefined,
      "styles": "width:100%;@media (min-width:640px){width:16.666666666666668%;};/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNvbC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUF3QlkiLCJmaWxlIjoiY29sLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgY3NzIH0gZnJvbSAnQGVtb3Rpb24vY29yZSc7XG5pbXBvcnQgY29uZmlnIGZyb20gJy4vR3JpZENvbmZpZyc7XG5pbXBvcnQgY29sdW1uV2lkdGggZnJvbSAnLi9jb2x1bW5XaWR0aCc7XG5pbXBvcnQgcmVzcG9uc2l2ZVdpZHRoIGZyb20gJy4vcmVzcG9uc2l2ZVdpZHRoJztcblxuY29uc3QgeyBjb2x1bW5OdW1iZXIgfSA9IGNvbmZpZztcblxuLypcbiAqIEBwYXJhbSB7bnVtYmVyfSBjb2x1bW4gd2lkdGggZXhwcmVzc2VkIGFzIGEgd2hvbGUgbnVtYmVyIGZyb20gMSB0byBtYXggQ29sdW1uc1xuICogQHJldHVybnMgeyhzdHJpbmd8bnVsbCl9IEVtb3Rpb25KUyBDU1Mgc3RyaW5nLCBvciBudWxsXG4gKiBAZXhhbXBsZVxuICogPENvbHVtbiBjc3M9e2NvbCgyKX0+XG4gKiA8Q29sdW1uIGNzcz17W2NvbCg2KSwgc29tZU90aGVyU3R5bGluZ119PlxuICogTk9URToganVzdCBiZWNhdXNlIHlvdSBDQU4gYWRkIGFkZGl0aW9uYWwgc3R5bGluZyB0byBhIGNvbHVtbiBkb2Vzbid0IG1lYW4geW91IFNIT1VMRC5cbiAqL1xuY29uc3QgY29sID0gKG51bWJlcikgPT4ge1xuICBpZiAoXG4gICAgbnVtYmVyID4gY29sdW1uTnVtYmVyXG4gICAgfHwgbnVtYmVyIDwgMVxuICAgIHx8IHR5cGVvZiBudW1iZXIgIT09ICdudW1iZXInXG4gICkge1xuICAgIHRocm93IG5ldyBFcnJvcihgVGhpcyBpcyBhICR7Y29sdW1uTnVtYmVyfSBjb2x1bW4gZ3JpZC4gXG4gICAgVGhlIHZhbHVlIHN1cHBsaWVkIHRvIGNvbCAoJHtudW1iZXJ9KSB3b24ndCB3b3JrIWApO1xuICB9XG4gIHJldHVybiBjc3NgXG4gICAgJHtyZXNwb25zaXZlV2lkdGgoY29sdW1uV2lkdGggKiBudW1iZXIpfTtcbiAgICBgO1xufTtcblxuZXhwb3J0IGRlZmF1bHQgY29sO1xuIl19 */",
    }
  }
/>
`;

exports[`Grid : col(n) method should throw an error when passed a non-number 1`] = `
"This is a 12 column grid. 
    The value supplied to col (oops) won't work!"
`;

exports[`Grid : col(n) method should throw an error when passed a number that is greater than the max number of grid columns 1`] = `
"This is a 12 column grid. 
    The value supplied to col (75) won't work!"
`;

exports[`Grid : col(n) method should throw an error when passed a number that is less than 1 1`] = `
"This is a 12 column grid. 
    The value supplied to col (0) won't work!"
`;
